src/program.ts

```diff
@@ -4353,6 +4357,25 @@ export class Class extends TypedElement {
     return null;
   }
 
+  /** Looks up the iterator for this class. */
+  lookupIterator(): Function | null {
+    var instance: Class | null = this;
+    do {
+      if (instance.members) {
+        for (let member of instance.members.values()) {
```

Comment:
Unlike TypeScript, which is unsound in this case, AssemblyScript cannot track that `instance.members` is non-null after the check for type safety reasons. Assigning `let members = instance.members` to a local, and then doing the check, solves this.